router/src/infer/v3/queue.rs

```diff
@@ -161,15 +178,25 @@ impl State {
         block_size: u32,
         window_size: Option<u32>,
         speculate: u32,
+        max_batch_total_tokens: u32,
     ) -> Self {
+        let block_allocator = match requires_padding {
```

Comment:
You can use [bool::then](https://doc.rust-lang.org/std/primitive.bool.html#method.then).